dataline-tests/src/acceptanceTests/java/io/dataline/test/acceptance/AcceptanceTests.java

```diff
@@ -75,61 +82,169 @@
+  @Test
+  public void testCreateSourceImplementation() throws ApiException {
+    String dbName = "acc-test-db";
+    UUID postgresSourceSpecId = getPostgresSourceSpecId();
+    UUID defaultWorkspaceId = PersistenceConstants.DEFAULT_WORKSPACE_ID;
+    Map<Object, Object> sourceDbConfig = getSourceDbConfig();
+    SourceImplementationRead response = createSourceImplementation(
+        dbName,
+        postgresSourceSpecId,
+        defaultWorkspaceId,
+        sourceDbConfig);
+
+    assertEquals(dbName, response.getName());
+    assertEquals(defaultWorkspaceId, response.getWorkspaceId());
+    assertEquals(postgresSourceSpecId, response.getSourceSpecificationId());
+    assertEquals(Jsons.jsonNode(sourceDbConfig), response.getConnectionConfiguration());
   }
 
-  @AfterAll
-  public static void tearDown() {
-    SOURCE_PSQL.stop();
-    TARGET_PSQL.stop();
+  @Test
+  public void testSourceCheckConnection() throws IOException, ApiException {
+    UUID sourceImplId = createPostgresSourceImpl().getSourceImplementationId();
+    CheckConnectionRead checkConnectionRead = apiClient.getSourceImplementationApi()
+        .checkConnectionToSourceImplementation(new SourceImplementationIdRequestBody().sourceImplementationId(sourceImplId));
+    assertEquals(CheckConnectionRead.StatusEnum.SUCCESS, checkConnectionRead.getStatus());
   }
 
   @Test
-  public void fullTestRun() throws IOException, ApiException, SQLException {
-    UUID createdSourceImplId = testCreateSourceImplementation().getSourceImplementationId();
-    testCheckSourceConnection(createdSourceImplId);
+  public void testDiscoverSourceSchema() throws ApiException, IOException {
+    UUID sourceImplementationId = createPostgresSourceImpl().getSourceImplementationId();
+    SourceSchema actualSchema = discoverSourceSchema(sourceImplementationId);
 
-    UUID createdDestinationImplId = testCreateDestinationImpl().getDestinationImplementationId();
-    testCheckDestinationConnection(createdDestinationImplId);
+    SourceSchema expectedSchema = Jsons.deserialize(MoreResources.readResource("simple_postgres_source_schema.json"), SourceSchema.class);
+
+    assertEquals(expectedSchema, actualSchema);
+  }
+
+  @Test
+  public void testCreateConnection() throws ApiException {
+    UUID sourceImplId = createPostgresSourceImpl().getSourceImplementationId();
+    SourceSchema schema = discoverSourceSchema(sourceImplId);
+    UUID destinationImplId = createPostgresDestinationImpl().getDestinationImplementationId();
+    String name = "test-connection-" + UUID.randomUUID().toString();
+    ConnectionSchedule schedule = new ConnectionSchedule().timeUnit(MINUTES).units(100L);
+    ConnectionCreate.SyncModeEnum syncMode = ConnectionCreate.SyncModeEnum.FULL_REFRESH;
+    ConnectionRead createdConnection = createConnection(name, sourceImplId, destinationImplId, schema, schedule, syncMode);
```

Comment:
In your tests can you separate the creation from the execution from the verification?

Reply:
done